File: 2026-08/src/day_29/prms_43162/Solution2.cpp
#include <bits/stdc++.h>
using namespace std;

const int MAXN = 200;
bool vis[MAXN];

void dfs(int cur, int n, vector<vector<int>>& computers) {
    vis[cur] = true;

    for (int nxt = 0; nxt < n; nxt++) {
        if (computers[cur][nxt] == 0 || vis[nxt]) continue;
        dfs(nxt, n, computers);
    }
}

int solution(int n, vector<vector<int>> computers) {
    int cnt = 0;

    for (int i = 0; i < n; i++) {
        if (vis[i]) continue;
        dfs(i, n, computers);
        cnt++;
    }

    return cnt;
}

File: 2026-08/src/day_29/prms_43162/Solution2.py
import sys

sys.setrecursionlimit(10**6)


def solution(n, computers):
    vis = [False] * n

    def dfs(cur):
        vis[cur] = True
        for nxt in range(n):
            if computers[cur][nxt] == 0 or vis[nxt]:
                continue
            dfs(nxt)

    cnt = 0
    for i in range(n):
        if vis[i]:
            continue
        dfs(i)
        cnt += 1

    return cnt
